processing/src/main/java/org/apache/druid/segment/CursorFactory.java

```diff
@@ -35,12 +38,55 @@
  */
 public interface CursorFactory
 {
+  default CursorMaker asCursorMaker(CursorBuildSpec spec)
+  {
+
+    return new CursorMaker()
+    {
+      @Override
+      public boolean canVectorize()
+      {
+        return CursorFactory.this.canVectorize(spec.getFilter(), spec.getVirtualColumns(), spec.isDescending());
```

Comment:
this is intended so that existing `CursorFactory`/`StorageAdapter` implementations can keep working with query engines without implementing `asCursorMaker`